packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison-node.vue

```diff
@@ -9,16 +9,22 @@
 import { watch, computed } from 'vue';
 import Button from 'primevue/button';
 import TeraOperatorPlaceholder from '@/components/operator/tera-operator-placeholder.vue';
-import { WorkflowNode, WorkflowPortStatus } from '@/types/workflow';
+import { WorkflowNode, WorkflowPortStatus, OperatorStatus } from '@/types/workflow';
 import { ModelComparisonOperationState } from './model-comparison-operation';
 
-const emit = defineEmits(['append-input-port', 'open-drilldown']);
+const emit = defineEmits(['append-input-port', 'open-drilldown', 'update-status']);
 
 const props = defineProps<{
 	node: WorkflowNode<ModelComparisonOperationState>;
 }>();
 
-const hasAtLeastTwoValues = computed(() => props.node.inputs.filter((input) => input.value).length >= 2);
+const hasAtLeastTwoValues = computed(() => {
+	const flag = props.node.inputs.filter((input) => input.value).length >= 2;
+	// This is a custom way of granting a default status to the operator, since it has no output
+	if (flag) emit('update-status', OperatorStatus.DEFAULT);
+	else emit('update-status', OperatorStatus.INVALID);
+	return flag;
```

Comment:
Generally speaking, I think we should avoid having computed properties having these types of "side effects" where it can modify external environments. It can cause unexpected behaviours since people do not tend to think of computed behaving in this manner.

Reply:
sure. I think it'll make more sense to set it to `DEFAULT` once the overview is generated